### PR TITLE
Update Youtube Downloader.py

### PR DESCRIPTION
**Pull request by Ashwin2105**
Added a button to configure proxy.
Added a Thumbnail Image display.
Changed Interface.
needs a Default.png image in the same directory as python script.
renamed variable names with prefix v and similarly for buttons, label, etc.
screenshot from 2018-12-27 22-46-59
screenshot from 2018-12-27 22-45-47
screenshot from 2018-12-27 22-46-36

### CODE REVIEW

**Comment by Ashwin2105** [comment]
Default.png must have width = 120px and height= 90px.
Here is what I used as Default.png
default

**Comment by shreyasbapat** [comment]
Amazing work @Ashwin2105 , I have a small question. What is `default.png` for?

File: YouTube Downloader.py [hunk 1]
```diff
@@ -2,16 +2,16 @@
 import tkinter as tk
 from tkinter import *
 from pytube import YouTube
-import pyperclip as pc 
+import pyperclip as pc
 import validators as vd
+from PIL import ImageTk, Image
+from io import BytesIO
+import requests
```
Comment (shreyasbapat) [inline YouTube Downloader.py]: The requests can be imported way avove every other thing for code quality related issues :)
```suggestion
import requests

import tkinter as tk
from tkinter import *

```

File: YouTube Downloader.py [hunk 2]
```diff
@@ -2,16 +2,16 @@
...
 
 root = tk.Tk()
-root.title('YouTube Video Downloader')
-Label(text='Put the link here in the box below: ').pack(side=TOP,padx=10,pady=10)
-
-qualityOptions = ['Click View options button']
-selectedOption = StringVar(root)
-# selectedOption.set('')
+root.title("Youtube Video Downloader")
+img = ImageTk.PhotoImage(Image.open("Default.png"))
```
Comment (shreyasbapat) [inline YouTube Downloader.py]: Same comment. WHat is `default.png` ?

File: YouTube Downloader.py [hunk 3]
```diff
@@ -20,52 +20,72 @@ def ret_copied_url():
 	else:
 		return 'https://www.youtube.com/watch?v=32HANv-bdJs'
 
-def callbackInput(sv):
+def callbackInput(v_url):
```
Comment (shreyasbapat) [inline YouTube Downloader.py]: Thanks a lot for this :)

File: YouTube Downloader.py [hunk 4]
```diff
@@ -20,52 +20,72 @@ def ret_copied_url():
...
+def changed(*options):
+    print('Changed option: ' + selectedOption.get())
+
+#variables
+qualityOptions = ["Select Quality"]
+selectedOption = StringVar(root)
+v_url = StringVar(root)
+my_proxy = { "http" : "", "https" : "" }
+
+v_url.trace("w", lambda name, index, mode, v_url=v_url: callbackInput(v_url))
```
Comment (shreyasbapat) [inline YouTube Downloader.py]: Clever!

File: YouTube Downloader.py [hunk 5]
```diff
@@ -20,52 +20,72 @@ def ret_copied_url():
...
+    
     for option in qualityOptions:
         qualityOptionsMenu['menu'].add_command(label=option, command=tk._setit(selectedOption, option))
     selectedOption.set(qualityOptions[0])
-
+    
+    #printing Video Thumbnail
+    t_url = video.thumbnail_url
+    response = requests.get(t_url)
+    img = Image.open(BytesIO(response.content))
```
Comment (shreyasbapat) [inline YouTube Downloader.py]: Great work!

File: YouTube Downloader.py [hunk 6]
```diff
@@ -78,28 +98,74 @@ def ondown():
         selectedStream.download()
         print('Selected video downloaded.')
 
+def Set_proxy_window():
```
Comment (shreyasbapat) [inline YouTube Downloader.py]: We can refactor the code a bit in upcoming PR's, but for now, its amazing. :)

File: YouTube Downloader.py [hunk 7]
```diff
@@ -78,28 +98,74 @@ def ondown():
...
+videoTitle.place(x=25,y=50,width=450)
+qualityOptionsMenu.place(x=25,y=75, width=180)
+B_setproxy.place(x=25,y=110)
+B_downlaod.place(x=115,y=110)
+B_exit.place(x=80,y=145)
+B_refresh.place(y=170,x=320)
+L_image.place(x=300,y=75,height=90,width=120)
+
+#setting root window size and position
+root.geometry("475x205+300+300")
```
Comment (shreyasbapat) [inline YouTube Downloader.py]: Again! This makes the app way cleaner. :)

File: YouTube Downloader.py (hunk 2 above)
Comment (shreyasbapat) [reply YouTube Downloader.py]: Oh! Okay

File: YouTube Downloader.py (hunk 1 above)
Comment (shreyasbapat) [reply YouTube Downloader.py]: @Ashwin2105  Just address this issue and it is merged.

**Comment by shreyasbapat** [comment]
Thanks a lot @Ashwin2105  for the contribution :rocket:

**Pull request merged**